### PR TITLE
where-are-the-logs

### LINKED ISSUE #368: Document where to find logs/make helper function
**Describe the new feature you'd like**
We should write up a quick doc page that describes where Holodeck logs are kept. It might also be nice to have a `holodeck.util.where_are_the_logs()` helper function or something that would print out the full path to the log directory, so users can find their logs without thinking too hard.

We should then link to this page in the issue template for opening a bug, so people can find it and include logs.

### PR DESCRIPTION
**Pull request by chris-luangrath**
A helper function that prints out the full path to the log directory so users can easily find the logs.

### CODE REVIEW

**Comment by chris-luangrath** [comment]
Right now the function returns 3 paths because I'm not sure which one is correct. Could someone help me with this? Other than that, I'm pretty sure that this should work correctly. Also, would a shorter name be better, like get_logs_path()?

**Comment by vinhowe** [comment]
On Linux, I've used logs at `~/.local/share/holodeck/0.3.0/worlds/DefaultWorlds/LinuxNoEditor/Holodeck/Saved/Logs/`. There may be more, not sure where to look for that.

**Comment by chris-luangrath** [comment]
> On Linux, I've used logs at `~/.local/share/holodeck/0.3.0/worlds/DefaultWorlds/LinuxNoEditor/Holodeck/Saved/Logs/`. There may be more, not sure where to look for that.

Thanks, I'll add that!

File: src/holodeck/util.py
Comment (vinhowe) [inline src/holodeck/util.py]: I don't have quick access to a Windows machine with Holodeck set up, so these notes might only apply to Linux:
- I think it would be cleaner to do
  ```python
  os.path.join(get_holodeck_path(), "worlds", .....)
  ```
  here.
- Since the user can have multiple world packages downloaded, and because DefaultWorlds isn't installed by default, I wonder if it would be better to use `packagemanager.installed_packages()` and iterate through each of them.
  - I don't know if the way we use our packages requires all the logs to be in the same place relative to the root of the package. My guess is yes, but you could look into that.

File: src/holodeck/util.py
Comment (vinhowe) [inline src/holodeck/util.py]: I don't really know what to do about hard-coding the UE version in the first Windows path. If we made a constant, how would we remember to update it when we upgraded UE? Maybe we could put it on our release checklist?

File: src/holodeck/util.py
Comment (vinhowe) [inline src/holodeck/util.py]: Do you think it's worth returning a list of strings instead of printing? Users could still print the paths if you returned a list, but they won't be able to programmatically use the values as-is. It would also be consistent with the other util functions.

File: .github/ISSUE_TEMPLATE/bug_report.md
Comment (vinhowe) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Seems like all this information could fit comfortably on one line

Comment (vinhowe) [inline .github/ISSUE_TEMPLATE/bug_report.md]: Seems like there's redundant information here. Maybe you could consolidate everything on the same line?

**Review by vinhowe — CHANGES_REQUESTED** [review]
This seems pretty useful. Looking at #368, would you be willing to write up a doc page with these paths and/or where to find this function--whatever seems like would be easiest for people? We could also do that in another PR if you'd prefer.

Also, I'd like to check the Windows paths and see if there's anywhere else logs end up, but I don't have Windows with Holodeck convenient right now. I'll have to circle back to that unless someone beats me to it.

File: src/holodeck/util.py
Comment (vinhowe) [inline src/holodeck/util.py]: It looks like the logs for the Holodeck engine itself (for DefaultWorlds) are at ` ~\AppData\Local\holodeck\0.3.0\worlds\DefaultWorlds\WindowsNoEditor\Holodeck\Saved\Logs` on Windows. Both points in my comment [here](https://github.com/BYU-PCCL/holodeck/pull/439/files#r554481608) also apply here.

**Comment by vinhowe** [comment]
@daniekpo maybe a second pair of eyes here would be better? What do you think about my comments?

File: src/holodeck/util.py
Comment (daniekpo) [reply src/holodeck/util.py]: Yeah, we should not be hardcoding the path. You want to call `get_holodeck_path` in `util.py` to get the install directory then navigate from there to the logs directory. The logs path is `[HOLODECK-DIR]/worlds/[OS]NoEditor/Holodeck/Saved/Logs`

File: src/holodeck/util.py
Comment (daniekpo) [reply src/holodeck/util.py]: You're right here. We may need to iterate through all the installed worlds and show the logs paths.

**Review by daniekpo — CHANGES_REQUESTED** [review]

File: src/holodeck/util.py
Comment (chris-luangrath) [reply src/holodeck/util.py]: I changed it so that it prints _and_ returns the path, does that work?

Comment (vinhowe) [reply src/holodeck/util.py]: @daniekpo what do you think here? Seems to me like the functional way would be to avoid side effects and just produce output.

Comment (daniekpo) [reply src/holodeck/util.py]: Yes. A function like this should typically just return the string (list of strings). The client/user might want to write it to a file or display it in the console. Just make it return the path(s) and don't worry about printing it out.

File: src/holodeck/util.py
Comment (vinhowe) [inline src/holodeck/util.py]: Seems like there's some duplication here, and this doesn't account for all worlds. Could we do something like this?
```python
return [
    os.path.join(
        get_holodeck_path(),
        "worlds",
        package,
        "{}NoEditor".format(get_os_key()),
        "Holodeck",
        "Saved",
        "Logs",
    )
    for package in holodeck.packagemanager.installed_packages()
]

```

`os.path.join` will join as many paths as you want, which you can use to avoid having to specify the entire path twice with Windows- and Unix-style slashes. There might be an even better way to do this--I haven't checked.

My example uses [list comprehension](https://www.w3schools.com/python/python_lists_comprehension.asp), but you could also do a `for` loop through the installed packages and just add them to an empty list.

**Review by vinhowe — CHANGES_REQUESTED** [review]

File: src/holodeck/util.py
Comment (chris-luangrath) [reply src/holodeck/util.py]: I tried this:
```python
    paths = []

    for package in holodeck.packagemanager.installed_packages():
        paths.append(
            os.path.join(
                get_holodeck_path(),
                'worlds',
                package,
                '{}NoEditor'.format(get_os_key()),
                "Holodeck",
                "Saved",
                "Logs",
            )
        )

    return paths
```
it returns: 
```
['C:\\Users\\Chris\\AppData\\Local\\holodeck\\0.3.2.dev0\\worlds\\DefaultWorlds\\WindowsNoEditor\\Holodeck\\Saved\\Logs']
```
it also brings this up when I call the function:
```
**********************************************
* You have old versions of Holodeck packages *
**********************************************
Use packagemanager.prune() to delete old packages
Versions: ['0.3.2.dev0 (old)']
Place an `ignore_old_packages` file in C:\Users\Chris\AppData\Local\holodeck to suppress this message
```
it that's alright, then I'll keep it like this

File: src/holodeck/util.py
Comment (chris-luangrath) [inline src/holodeck/util.py]: Black says that it would reformat this file, but I'm not sure what it'd want to change. Is there a way to find that?

Comment (daniekpo) [reply src/holodeck/util.py]: You can run black and give it the file name as a parameter. Alternatively, you can make black your formatter in vs code and run the format code command.

File: src/holodeck/util.py
Comment (vinhowe) [inline src/holodeck/util.py]: Could you rename this to `log_paths`? I think that makes more sense. Once you get everything formatted, LGTM!

**Review by vinhowe — CHANGES_REQUESTED** [review]

**Pull request merged**
